src/libraries/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -170,6 +170,21 @@ public void TrimExcess()
             }
         }
 
+        /// <summary>
+        /// Sets the capacity of a <see cref="Stack{T}"/> object to a specified number of entries.
+        /// </summary>
+        /// <param name="capacity">The new capacity.</param>
+        public void TrimExcess(int capacity)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(capacity);
+            ArgumentOutOfRangeException.ThrowIfLessThan(capacity, _size);
+
+            if (capacity == _size)
```

Comment:
Shouldn't this have been `if (capacity == _array.Length)`? With the current implementation calling `stack.TrimExcess(stack.Count)` would be a no-op.